src/Disks/IO/IOUringReader.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#if defined(OS_LINUX)
+
+#include <Common/ThreadPool.h>
+#include <IO/AsynchronousReader.h>
+#include <unordered_map>
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+
+#pragma clang diagnostic ignored "-Wold-style-cast"
+#pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
+
+#ifdef HAS_RESERVED_IDENTIFIER
+#pragma clang diagnostic ignored "-Wreserved-identifier"
+#endif
+
+#endif
+
+#include <liburing.h>
```

Comment:
You can include this header as `SYSTEM` (in `target_include_directories` in CMake). It will inhibit warnings (similar to the case if it is located in /usr/include).